pkg/prometheus/promcfg.go

```diff
@@ -938,7 +938,9 @@ func (cg *ConfigGenerator) generateProbeConfig(
 		xc := labeler.GetRelabelingConfigs(m.TypeMeta, m.ObjectMeta, m.Spec.Targets.StaticConfig.RelabelConfigs)
 		relabelings = append(relabelings, generateRelabelConfig(xc)...)
 		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: relabelings})
-	} else {
+	}
+
+	if m.Spec.Targets.Ingress != nil {
```

Comment:
The doc says that static takes precedence over ingress if both are defined.

```suggestion
	} else if m.Spec.Targets.Ingress != nil {
```